Sources/WalletSdk/MDoc.swift

```diff
@@ -128,25 +128,42 @@ extension BLESessionManager: MDocBLEDelegate {
                 let req = requestData.itemsRequests
                 self.callback.update(state: .selectNamespaces(req))
             } catch {
-                self.callback.update(state: .error("\(error)"))
+                self.callback.update(state: .error(.generic("\(error)")))
                 self.cancel()
             }
         case .error(let error):
-            self.callback.update(state: .error("\(error)"))
+            self.callback.update(state: .error(BleSessionError(holderBleError: error)))
             self.cancel()
         }
     }
 }
 
+public enum BleSessionError {
+    case bleStack(String)
+    case unauthorized(String)
```

Comment:
I think it would be worth returning `central.state` directly, in the interest of avoiding any language and to let the app have fine-grained control over it